cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java

```diff
@@ -270,11 +270,11 @@ private void tagsQuerying(HttpRequest request, HttpResponder responder, List<Str
   }
 
   private void contextQuerying(HttpRequest request, HttpResponder responder,
-                               String context, String metric, String groupBy) {
+                               String context, List<String> metrics, String groupBy) {
     try {
       List<String> groupByTags = parseGroupBy(groupBy);
       MetricQueryResult queryResult = executeQuery(request, parseTagValuesAsMap(context),
-                                                   groupByTags, ImmutableList.of(metric));
+                                                   groupByTags, metrics);
```

Comment:
what does executeQuery() do if metrics is null? It is noyt annotated as @Nullable, and neither are the corresponding parameters of methods called from there. Are you sure it is ok to send null?